### PR TITLE
Ord 2

### PR DESCRIPTION
**Pull request by kabirbaidhya**
Item CRUD

### CODE REVIEW

File: module/Order/src/Order/Controller/ItemController.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    public function indexAction()
+    {
+        $request = $this->getRequest();
+        $query = $request->getQuery();
+
+        $max = (int)$query->get('max', 10);
+        $page = (int)$query->get('page', 1);
+
+        if ($page < 1) {
+            throw new BadRequestException('Invalid page');
```
Comment (lf-kailashraj) [inline module/Order/src/Order/Controller/ItemController.php]: use translation here.

File: module/Order/src/Order/Entity/Item.php [hunk 2]
```diff
@@ -30,6 +30,12 @@ class Item {
      **/
     protected $rate;
 
+    public function __construct()
+    {
+        $this->created = new DateTime();
```
Comment (lf-kailashraj) [inline module/Order/src/Order/Entity/Item.php]: i think pre/post persist methods are preferred.
http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html#prepersist

File: module/Order/src/Order/Entity/ItemRepository.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+namespace Order\Entity;
+
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Tools\Pagination\Paginator;
+
+class ItemRepository extends EntityRepository
+{
+    public function fetchList($offset = 0, $max = 10)
```
Comment (lf-kailashraj) [inline module/Order/src/Order/Entity/ItemRepository.php]: constant values should be avoided strictly. create a static const somewhere.

**Comment by kabirbaidhya** [comment]
Ok

**Comment by mesaugat** [comment]
:+1:

**Pull request merged**